magic-modules/generate_terraform.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+# Setup GOPATH
+export GOPATH="${PWD}/go"
+WORKDIR="${PWD}"
+
+set -x
+set -e
+
+# Create GOPATH structure
+mkdir -p "${GOPATH}/src/github.com/terraform-providers"
+
+cd magic-modules
+git submodule update --init build/terraform
+ln -s "${PWD}/build/terraform/" "${GOPATH}/src/github.com/terraform-providers/terraform-provider-google"
+
+cd "${GOPATH}/src/github.com/terraform-providers/terraform-provider-google"
+
+go get
+
+cd "${WORKDIR}/magic-modules"
+BRANCH=$(git rev-parse --short HEAD)
+
+bundle install
+bundle exec ruby compiler.rb -p products/compute -e terraform -o "${GOPATH}/src/github.com/terraform-providers/terraform-provider-google/"
+
+cd "build/terraform"
+git add -A
+git config --global user.email "nmckinley@google.com"
+git config --global user.name "Magic Module Bot"
+git commit -m "magic modules change happened here" || true  # don't crash if no changes
+git checkout -B $BRANCH
+
+cd "../../"
+git config -f .gitmodules submodule.build/terraform.branch $BRANCH
+git config -f .gitmodules submodule.build/terraform.url "git@github.com:$GH_USERNAME/terraform-provider-google.git"
+git submodule sync
```

Comment:
In some parts we do actions to build/terraform only. Here we do for all modules. Pick a strategy for them all? (my favorite would be to deal with just the submodule being operated at the time.

Reply:
That's on purpose - the idea is that the parts that are done for all submodules will stay the same for every submodule, and the parts for terraform only will be copied / modified for other submodules.

Reply:
Sure, changed it.